qe/drivers.py

```diff
@@ -1934,28 +1902,24 @@ def __init__(
         psi_j: Psi_det,
         d_one_e_integral: One_electron_integral,
         E0: Energy,
-        comm: MPI.COMM_WORLD = None,
+        comm=MPI.COMM_WORLD,
     ):
         self.d_one_e_integral = d_one_e_integral
         self.E0 = E0
-        if comm is None:  # Non-MPI jobs
-            self.comm = None
+        self.comm = comm
+        self.world_size = self.comm.Get_size()  # No. of processes running
+        self.rank = self.comm.Get_rank()  # Rank of current process
+        self.MPI_master_rank = 0  # Master rank
+        # Dispatch wavefunction and save psi_local, dictionary of 2e integrals with local instance of class
+        if self.world_size == 1:  # One process running, use entire wavefunction
+            self.psi_i = psi_j
+        else:  # Else, dispatch chunks of the wavefunction to all processes
             self.psi_i = psi_i
```

Comment:
Why do this check here?
In theory, when using 1 rank, we still have `chunk` , just one :) 
I would prefer to keep this kind of logic out-side of the function if needed